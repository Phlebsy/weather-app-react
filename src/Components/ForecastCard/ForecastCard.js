import React from 'react';
import PropTypes from 'prop-types';
import { convertFromKelvin, getIcon } from '../../utilities/utilities';

// ForecastCard represents the tri-hourly forecasts for a day
// Logic is passed in from parent component, in order to change the active forecast
export default function ForecastCard(props) {
  const { forecast, onClick } = props;
  return (
    <div
      onClick={onClick}
      className="d-flex flex-column align-items-center py-3 bg-warning "
    >
      <i
        className={
          'fas fa-' + getIcon(forecast['dateTime'], forecast['weather'][0])
        }
      />
      <p className="m-0">{convertFromKelvin(forecast['main']['temp'])}&deg;</p>
      <p className="m-0">{forecast['dateTime']}</p>
    </div>
  );
}

ForecastCard.propTypes = {
  forecast: PropTypes.object.isRequired,
  onClick: PropTypes.func
};
