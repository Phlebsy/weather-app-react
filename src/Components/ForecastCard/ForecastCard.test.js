import React from 'react';
import ReactDOM from 'react-dom';
import ForecastCard from './ForecastCard';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const forecastObj = {
  dateTime: '6:15',
  weather: [
    {
      description: 'clear sky',
      main: 'Clear'
    }
  ],
  main: {
    temp: 276.68,
    pressure: 1010,
    humidity: 60,
    temp_min: 275.95,
    temp_max: 277.55
  },
  wind: {
    speed: 4.1,
    deg: 320
  }
};

describe('<ForecastCard />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ForecastCard forecast={forecastObj} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders icon correctly', () => {
    const wrapper = shallow(<ForecastCard forecast={forecastObj} />);
    expect(wrapper.find('.fa-sun').length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<ForecastCard forecast={forecastObj} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
