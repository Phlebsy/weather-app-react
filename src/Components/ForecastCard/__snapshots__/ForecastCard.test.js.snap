// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastCard /> matches the snapshot 1`] = `
<div
  className="d-flex flex-column align-items-center py-3 bg-warning "
>
  <i
    className="fas fa-sun"
  />
  <p
    className="m-0"
  >
    38
    Â°
  </p>
  <p
    className="m-0"
  >
    6:15
  </p>
</div>
`;
